### PR TITLE
test: CON-1433 Enable `cup_compatibility_test` for new `NiDkgTag::HighThresholdForKey`

### PR DESCRIPTION
**Pull request by eichhorl**
The mainnet version now support the new `NiDkgTag::HighThresholdForKey`, therefore we can start creating artifacts containing these tags as part of the `cup_compatibility_test`.

### CODE REVIEW

File: rs/types/types/src/exhaustive.rs [hunk 1]
```diff
@@ -632,28 +664,41 @@ impl<T: ExhaustiveSet> ExhaustiveSet for Signed<T, MultiSignature<T>> {
...
-    fn exhaustive_set<R: RngCore + CryptoRng>(_: &mut R) -> Vec<Self> {
-        vec![NiDkgConfig::new(valid_dkg_config_data()).unwrap()]
+    fn exhaustive_set<R: RngCore + CryptoRng>(rng: &mut R) -> Vec<Self> {
+        let nidkg_ids = NiDkgId::exhaustive_set(rng);
+        nidkg_ids
+            .into_iter()
+            .map(|id| {
+                let mut config_data = valid_dkg_config_data();
+                config_data.dkg_id = id.clone();
+                config_data.resharing_transcript.as_mut().unwrap().dkg_id = id;
```
[the remark this answers is not in the document]
Comment (eichhorl) [reply rs/types/types/src/exhaustive.rs]: Good point, I think I confused this with the implementation below. Instead we should test the case where the `resharing_transcript` is `None`.

File: rs/types/types/src/exhaustive.rs [hunk 2]
```diff
@@ -19,6 +19,7 @@ use crate::crypto::canister_threshold_sig::idkg::{
     IDkgTranscriptId, IDkgTranscriptOperation, IDkgTranscriptParams, IDkgTranscriptType,
     InitialIDkgDealings, SignedIDkgDealing,
 };
+use crate::crypto::threshold_sig::ni_dkg::NiDkgMasterPublicKeyId;
```
Comment (eichhorl) [inline rs/types/types/src/exhaustive.rs:22]:
```suggestion
```

File: rs/types/types/src/exhaustive.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (fspreiss) [reply rs/types/types/src/exhaustive.rs]: 👍

File: rs/types/types/src/exhaustive.rs [hunk 3]
```diff
@@ -375,15 +382,45 @@ impl ExhaustiveSet for SchnorrKeyId {
...
+            .chain(vetkd_key_ids.into_iter().map(MasterPublicKeyId::VetKd))
+            .collect()
+    }
+}
+
+impl ExhaustiveSet for IDkgMasterPublicKeyId {
+    fn exhaustive_set<R: RngCore + CryptoRng>(rng: &mut R) -> Vec<Self> {
+        MasterPublicKeyId::exhaustive_set(rng)
+            .into_iter()
+            .flat_map(IDkgMasterPublicKeyId::try_from)
```
Comment (Sawchord) [inline rs/types/types/src/exhaustive.rs:423]: NIT:
I know that this is much shorter and concise, but I find the fact that `Result` implements `IntoIterator` and that you can use that to use `flat_map` to filter out errors kinda cursed.

Even though it is quite a bit longer, I would use a `filter_map` and match on the results.

**Review by Sawchord — APPROVED** [review]
LGTM

**Pull request merged**